### ISSUE TITLE
Logger 

### ISSUE THREAD

**Issue by to92me** [bstaletic/eurobot2k17#16]
propose API and features and implement it.

**Comment by to92me** [comment]
propose:

1. ```error(String _error);``` 
2. ```debug(String _dubug);``` 
3. ```printf(...);```

on Error turn on red light somewhere. 
redirect printf to UART

**Comment by bstaletic** [comment]
Should we keep the red light on forever once an error occurs? I think it could be useful to turn it off at some point, so that new errors would be easily noticable. Though, I can't figure out when would be the right time to turn the light off.

**Comment by ivar94** [comment]
@to92me that pretty much covers it. I might add that i like the idea of turning a red diode! or we could stick a little speaker on robot and have it yell "KVRAGU" when error occurs?
@bstaletic turn it off when you again set robot to motion, or similar actions?

**Comment by bstaletic** [comment]
>turn it off when you again set robot to motion, or similar actions

Maybe, but that does in no way make sure that we saw the error in time.

**Comment by to92me** [comment]
>>turn it off when you again set robot to motion, or similar actions

>Maybe, but that does in no way make sure that we saw the error in time.

I agree there are no guarantees that we will see the red light. Some timer and turn off light after 5 seconds ?

**Comment by bstaletic** [comment]
5 seconds sounds like a reasonably long time. If others agree I'd be okay with that solution.

**Comment by to92me** [comment]
ok, then API is: 
1. ```error(String _error);```  writes data with printf() and turns on red light for 5 seconds. 
2. ```debug(String _dubug);``` if macro LOGGER_DEBUG is set to 1 prints data with printf()
3. ```printf(...);``` prints data to empty UART line. @darko31 which UART is free for this job ?

**Comment by bstaletic** [comment]
We have USART2, USART3 and UART4 configured.
USART2 is in halfduplex mode so we don't need a tri-state buffer for AX12.
USART3 is used in motion.c to communicate with motion driver board.
That only leaves UART4.

**Comment by to92me** [comment]
new propose: 
1. 
```C
error(const char* format, ... )
```    
error output with red led for 5 seconds 
2.
 ```C
debug(const char* format, ... )
 ```
 debug output if macro LOGGER_DEBUG_OUTPUT is set to 1 
3. 
```C 
info(const char* format, ... )
 ```
 info output if macro  LOGGER_INFO_OUTPUT is set to 1 

Output is UART4 

*note: parameter list ( const char* format, ... ) is  same as ```printf```'s parameter list, so you can use the same syntax

**Comment by bstaletic** [comment]
I love the idea of `printf` syntax, but I have one suggestion.

Should we maybe use `LOGGER_DEBUG` insteadd of `LOGGER_DEBUG_OUTPUT` and the same for `LOGGER_INFO`. Reason behind my suggestion would be a shorter macro, so not actually a big deal.

**Comment by to92me** [comment]
@bstaletic no problem, macros will be shortened :)

**Comment by to92me** [comment]
done without red led on error. This will be new task. Merged on e5555de
